src/Serialization.purs

```diff
@@ -0,0 +1,187 @@
+foreign import baseAddressPaymentCredential :: BaseAddress -> StakeCredential
+foreign import baseAddressToAddress :: BaseAddress -> Effect Address
+foreign import baseAddressFromAddress :: Address -> Effect BaseAddress
+foreign import newStakeCredentialFromScriptHash :: ScriptHash -> Effect StakeCredential
+foreign import newStakeCredentialFromKeyHash :: Ed25519KeyHash -> Effect StakeCredential
+foreign import newEd25519KeyHashFromBech32 :: T.Bech32 -> Effect Ed25519KeyHash
+foreign import newMultiAsset :: Effect MultiAsset
+foreign import insertMultiAsset :: MultiAsset -> ScriptHash -> Assets -> Effect Unit
+foreign import newAssets :: Effect Assets
+foreign import insertAssets :: Assets -> AssetName -> BigNum -> Effect Unit
+foreign import newAssetName :: ByteArray -> Effect AssetName
+foreign import newScriptHashFromBech32 :: T.Bech32 -> Effect ScriptHash
+foreign import newScriptHash :: ByteArray -> Effect ScriptHash
+foreign import newDataHash :: ByteArray -> Effect DataHash
+foreign import transactionOutputSetDataHash :: TransactionOutput -> DataHash -> Effect Unit
+foreign import newVkeywitnesses :: Effect Vkeywitnesses
+foreign import newVkeywitness :: Vkey -> Ed25519Signature -> Effect Vkeywitness
+foreign import addVkeywitness :: Vkeywitnesses -> Vkeywitness -> Effect Unit
+foreign import newVkeyFromPublicKey :: PublicKey -> Effect Vkey
+foreign import newPublicKey :: T.Bech32 -> Effect PublicKey
+foreign import newEd25519Signature :: T.Bech32 -> Effect Ed25519Signature
+foreign import transactionWitnessSetSetVkeys :: TransactionWitnessSet -> Vkeywitnesses -> Effect Unit
+foreign import newPlutusScript :: ByteArray -> Effect PlutusScript
+foreign import newPlutusScripts :: Effect PlutusScripts
+foreign import txWitnessSetSetPlutusScripts :: TransactionWitnessSet -> PlutusScripts -> Effect Unit
+foreign import addPlutusScript :: PlutusScripts -> PlutusScript -> Effect Unit
+foreign import addressPubKeyHashImpl :: (forall a. a -> Maybe a) -> (forall a. Maybe a) -> BaseAddress -> Maybe T.Bech32
+foreign import toBytes
+  :: ( Transaction
+   |+| TransactionOutput
+   -- Add more as needed.
+  ) -> Effect ByteArray
+
+addressPubKeyHash :: BaseAddress -> Maybe T.Bech32
+addressPubKeyHash = addressPubKeyHashImpl Just Nothing
+
+newBaseAddressFromAddress :: Address -> Maybe BaseAddress
+newBaseAddressFromAddress = newBaseAddressFromAddressImpl Nothing Just
+
+convertTransaction :: T.Transaction -> Effect Transaction
+convertTransaction (T.Transaction { body: T.TxBody body, witness_set }) = do
+  inputs <- convertTxInputs body.inputs
+  outputs <- convertTxOutputs body.outputs
+  fee <- convertBigInt (unwrap body.fee)
+  txBody <- newTransactionBody inputs outputs fee
+  ws <- convertWitnessSet witness_set
+  newTransaction txBody ws
+
+convertBigInt :: BigInt.BigInt -> Effect BigNum
+convertBigInt = newBigNum <<< BigInt.toString
+
+-- Parts that are commented-out are unused & untested, but should compile
```

Comment:
Are you planning on completing and testing these before merging the PR?

Reply:
Not before. I accidentally wrote more code than was needed. This may save someone some time once we need it. There is some commented-out code in `Transaction.purs`, so I thought that the practice of leaving dead code commented-out is OK.